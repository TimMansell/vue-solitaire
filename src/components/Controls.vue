<template>
  <div class="controls">
    <button class="controls__btn" @click="restartGame" data-test="new-game">
      New Game
    </button>
  </div>
</template>

<script>
export default {
  name: 'Controls',
  methods: {
    restartGame() {
      this.$store.dispatch('restartGame');
      this.$store.dispatch('initGame');
    },
  },
};
</script>

<style lang="scss" scoped>
.controls {
  position: fixed;
  right: 0.5rem;
  top: 0.5rem;

  &__btn {
    font-size: 1rem;
  }
}
</style>
