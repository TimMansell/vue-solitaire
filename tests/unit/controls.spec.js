import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Controls from '@/components/Controls.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Controls.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Controls);

    expect(wrapper).toMatchSnapshot();
  });

  it('calls store action "moveCardToControls" when clicked', () => {
    const actions = {
      restartGame: jest.fn(),
      initGame: jest.fn(),
    };

    const store = new Vuex.Store({
      actions,
    });

    const wrapper = shallowMount(Controls, { store, localVue });

    wrapper.find('[data-test="new-game"]').trigger('click');

    expect(actions.restartGame).toHaveBeenCalled();
    expect(actions.initGame).toHaveBeenCalled();
  });
});
