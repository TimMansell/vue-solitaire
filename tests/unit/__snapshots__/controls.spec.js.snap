// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls.vue matches snapshot 1`] = `
<div class="controls">
  <newgame-stub></newgame-stub>
  <rules-stub></rules-stub>
</div>
`;
